kover-gradle-plugin/src/functionalTest/kotlin/kotlinx/kover/gradle/plugin/test/functional/cases/ReportInheritedFromFilterTests.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.gradle.plugin.test.functional.cases
+
+import kotlinx.kover.gradle.plugin.test.functional.framework.configurator.BuildConfigurator
+import kotlinx.kover.gradle.plugin.test.functional.framework.starter.GeneratedTest
+
+internal class ReportInheritedFromFilterTests {
+    @GeneratedTest
+    fun BuildConfigurator.testExclusions() {
+        addProjectWithKover {
+            sourcesFrom("inherited-main")
+            kover {
+                reports {
+                    filters {
+                        excludes {
+                            inheritedFrom("*.Interface", "org.jetbrains.A", "*AutoCloseable")
+                        }
+                    }
+                }
+            }
+        }
+
+        run("koverXmlReport") {
+            xmlReport {
+                classCounter("org.jetbrains.RegularClass").assertFullyMissed()
+                classCounter("org.jetbrains.CloseableClass").assertFullyMissed()
+
+                classCounter("org.jetbrains.B").assertAbsent()
+                classCounter("org.jetbrains.C").assertAbsent()
+                classCounter("org.jetbrains.D").assertAbsent()
+                classCounter("org.jetbrains.AChild").assertAbsent()
+                classCounter("org.jetbrains.BChild").assertAbsent()
+                classCounter("org.jetbrains.CChild").assertAbsent()
+                classCounter("org.jetbrains.DChild").assertAbsent()
+            }
+        }
+    }
+    @GeneratedTest
+    fun BuildConfigurator.testInclusions() {
+        addProjectWithKover {
+            sourcesFrom("inherited-main")
+            kover {
+                reports {
+                    filters {
+                        includes {
+                            inheritedFrom("*.Interface", "org.jetbrains.A", "*AutoCloseable")
+                        }
+                    }
+                }
+            }
+        }
+
+        run("koverXmlReport") {
+            xmlReport {
+                classCounter("org.jetbrains.RegularClass").assertAbsent()
+                classCounter("org.jetbrains.CloseableClass").assertAbsent()
```

Comment:
Why it is absent, if Closable implements AutoClosable?

UPD: Ah, I suppose it's because documentation says `however they are directly inherited or implemented by any type from the application`. It is a kinda puzzling behavior. Is there any reason for it?

Reply:
Because implementation details: reporter works not in runtime - it takes class files from disk, and analyze it by ASM. If type is not present in project output, reporter can't can't find out which classes and interfaces are used in it.